pkg/recorder/recorder.go

```diff
@@ -85,7 +85,7 @@ func (r *Recorder) Record(rec record.Record) error {
 		return fmt.Errorf("record %s(size=%d) exceeds the archive size limit %d and will not be included in the archive",
 			recordName, recordSize, r.maxArchiveSize)
 	}
-	r.records[recordName] = memoryRecord
+	r.records[memoryRecord.Name] = memoryRecord
```

Comment:
@tremes I think that this thing could be a problem, 'cause the way that we flush things. You don't want to use the `memoryRecord.Name` you should definitely use the `recordName`. There are other implications to change this line.

/remove-lgtm

Reply:
Just one example, that you can see down below, the function `has` checks it by the Filename, not the RecordName. There is a reason to use the `Filename` instead of the `Name`, but TBH I don't record right now, I think that it is related to the data struct between the `MemoryRecord` and the `Record`.